File: app/src/main/java/org/kinecosystem/kinit/SendKinService.java
package org.kinecosystem.kinit;

import android.support.annotation.NonNull;

import org.kinecosystem.appsdiscovery.sender.service.SendKinServiceBase;

import java.math.BigDecimal;


public class SendKinService extends SendKinServiceBase {

    @Override
    public @NonNull
    KinTransferComplete transferKin(@NonNull String toAddress, int amount, @NonNull String memo) throws KinTransferException {
//        SampleWallet sampleWallet = ((SenderApplication) getApplicationContext()).getSampleWallet();
//        String sourceAddress = "None";
//
//        if (!sampleWallet.hasActiveAccount()) {
//
//            throw new KinTransferException(sourceAddress, "Cannot transfer Kin. Account not initialized");
//        }
//
       try {
//            sourceAddress = sampleWallet.getAccount().getPublicAddress();
//
//            int fee = 100; // no whitelisting for sample app, so using a fee
//            Transaction transaction = sampleWallet.getAccount().buildTransactionSync(toAddress,
//                    new BigDecimal(amount), fee, memo);
//            TransactionId transactionId = sampleWallet.getAccount().sendTransactionSync(transaction);
//
//            // here you may add some code to add the transaction details to
//            // transaction history metadata

            return new KinTransferComplete("my address", "transid");

        } catch (Exception e) {

            e.printStackTrace();

            throw new KinTransferException("my addresse2",
                    "Cannot transfer Kin. Exception " + e + ", with message " + e.getMessage());

        }
    }

    @Override
    public BigDecimal getCurrentBalance() throws BalanceException {

        try {
            return new BigDecimal(555);

        } catch (Exception e) {
            throw new BalanceException("Unable to retrieve Kin balance. Exception " + e + ", with message " + e.getMessage());
        }

    }
}

File: app/src/main/java/org/kinecosystem/kinit/kindiscover/SendKinService.kt
package org.kinecosystem.kinit.kindiscover

import org.kinecosystem.kinit.KinitApplication
import org.kinecosystem.kinit.server.NetworkServices
import org.kinecosystem.kinit.server.OperationResultCallback
import org.kinecosystem.transfer.repositories.KinTransferCallback
import org.kinecosystem.transfer.sender.service.SendKinServiceBase
import java.math.BigDecimal
import javax.inject.Inject

class SendKinService : SendKinServiceBase() {

    @Inject
    lateinit var networkServices: NetworkServices
    val address: String

    init {
        KinitApplication.coreComponent.inject(this)
        address = networkServices.walletService.userRepo.userInfo.publicAddress
    }

    @Throws(KinTransferException::class)
    override fun transferKin(toAddress: String, amount: Int, memo: String): KinTransferComplete? {
        return null
    }

    override fun transferKinAsync(toAddress: String, amount: Int, memo: String, callback: KinTransferCallback) {
        networkServices.offerService.app2appTransfer(toAddress, "14", amount, object : OperationResultCallback<String> {
            override fun onResult(result: String) {
                callback.onSuccess(KinTransferComplete(address, result, "some memo"))
            }

            override fun onError(errorCode: Int) {
                callback.onError(KinTransferException(address, "some error with error code $errorCode"))
            }
        })
    }

    @Throws(BalanceException::class)
    override fun getCurrentBalance(): BigDecimal {
        val balanceInt = networkServices.offerService.wallet.balanceInt
        return BigDecimal(balanceInt)
    }
}
